### PR TITLE
Fixes the color of the dropdown arrow in weird themes

### PR DESCRIPTION
**Pull request by zadjii-msft**
See also:
https://github.com/microsoft/microsoft-ui-xaml/blob/33732458ede38166f9127675898e4760c82b89bb/dev/SplitButton/SplitButton.xaml#L290-L293

We need to 
* Also set `SplitButtonForegroundSecondary` cause SplitButton's use that resource separately from the `Foreground()` property
* Manually trigger the visual state change, to refresh the brushes. We do something similar in TabBase

This is one of the bullet points in #13725

### CODE REVIEW

**Comment by DHowett** [comment]
I am quite scared that this changes fifteen files for what sounds like it shouldn't be too hard!

**Comment by DHowett** [comment]
Oh, I think you need to merge _`main`_ to comport the history. Otherwise it will continue to show changes that are already on main!

**Comment by zadjii-msft** [comment]
(diff should be better now)

**Comment by DHowett** [comment]
Moved from body of PR (so it didn't clutter up the git commit history with  HTML/JSON things):

Tested with these two themes:

<details>
<summary>json</summary>


```json
        {
            "name": "horrible",
            "tab":
            {
                "background": "#00FF00FF",
                "showCloseButton": "never"
            },
            "tabRow":
            {
                "background": "accent",
                "unfocusedBackground": "#008888FF"
            },
            "window":
            {
                "applicationTheme": "light",
                "useMica": false
            }
        },
        {
            "name": "horrible dark",
            "tab":
            {
                "background": "#00FF00FF",
                "showCloseButton": "never"
            },
            "tabRow":
            {
                "background": "accent",
                "unfocusedBackground": "#008888FF"
            },
            "window":
            {
                "applicationTheme": "dark",
                "useMica": false
            }
        },
```


</details>

File: src/cascadia/TerminalApp/TerminalPage.cpp [hunk 1]
```diff
@@ -3092,12 +3092,24 @@ namespace winrt::TerminalApp::implementation
         _newTabButton.Resources().Insert(winrt::box_value(L"SplitButtonBackgroundPointerOver"), backgroundHoverBrush);
         _newTabButton.Resources().Insert(winrt::box_value(L"SplitButtonBackgroundPressed"), backgroundPressedBrush);
 
+        // Load bearing: The SplitButton uses SplitButtonForegroundSecondary for
+        // the secondary button, but {TemplateBinding Foreground} for the
```
Comment (DHowett) [inline src/cascadia/TerminalApp/TerminalPage.cpp:3096]: yikes can we file that on them

**Review by DHowett — APPROVED** [review]

**Review by carlos-zamora — APPROVED** [review]

**Comment by carlos-zamora** [comment]
@msftbot merge this in 10 minutes

**Pull request merged**